app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,363 @@
+        final String newWord = mWordEditText.getText().toString();
+        final String newShortcut;
+        final String newWeight;
+
+        if (null == mShortcutEditText) {
+            newShortcut = null;
+        } else {
+            final String tmpShortcut = mShortcutEditText.getText().toString();
+            if (TextUtils.isEmpty(tmpShortcut)) {
+                newShortcut = null;
+            } else {
+                newShortcut = tmpShortcut;
+            }
+        }
+
+        if (mWeightEditText == null) {
+            newWeight = String.valueOf(WEIGHT_FOR_USER_DICTIONARY_ADDS);
+        } else {
+            final String tmpWeight = mWeightEditText.getText().toString();
+            if (TextUtils.isEmpty(tmpWeight)) {
+                newWeight = String.valueOf(WEIGHT_FOR_USER_DICTIONARY_ADDS);
+            } else {
+                newWeight = tmpWeight;
+            }
+        }
+
+        if (TextUtils.isEmpty(newWord)) {
+            // If the word is empty, don't insert it.
+            return CODE_CANCEL;
+        }
+
+        mSavedWord = newWord;
+        mSavedShortcut = newShortcut;
+        mSavedWeight = newWeight;
+
+        // In edit mode, everything is modified without overwriting other existing words
+        if (MODE_EDIT == mMode && hasWord(newWord, context) && newWord.equals(mOldWord)) {
+            UserDictionarySettings.deleteWordInEditMode(mOldWord, mOldShortcut, mOldWeight, resolver);
+            // Toast appears to indicate that the word has been modified
+            Toast.makeText(context, context.getText(R.string.user_dict_word_modified), Toast.LENGTH_SHORT).show();
+        } else {
+            mMode = MODE_INSERT;
+        }
+
+        // If the word already exists, a dialog box appears asking you to change the word
+        // See UserDictionaryAddWordFragment.addWord()
+        if (mMode == MODE_INSERT && hasWord(newWord, context)) {
+            return CODE_ALREADY_PRESENT;
+        }
+
+        if (mMode == MODE_INSERT) {
+            // Delete duplicate when adding or updating new word
+            UserDictionarySettings.deleteWordInEditMode(mOldWord, mOldShortcut, mOldWeight, resolver);
+            // Update the existing word by adding a new one
+            UserDictionary.Words.addWord(context, newWord,
+                    Integer.parseInt(newWeight), newShortcut, TextUtils.isEmpty(mLocale) ?
+                            null : LocaleUtils.constructLocaleFromString(mLocale));
+
+            // Toast appears either to indicate that the word has been modified or created
+            if (!TextUtils.isEmpty(mOldWord)) {
+                Toast.makeText(context, context.getText(R.string.user_dict_word_modified), Toast.LENGTH_SHORT).show();
+            } else {
+                final String localeInToast = new LocaleRenderer(context, mLocale).toString();
+                final String messageWordAdded = context.getString(R.string.user_dict_word_added) + " " + localeInToast;
```

Comment:
Don't concatenate strings for display, this can be really awkward in some languages.
Better use `context.getString(R.string.user_dict_word_added, localeInToast)` and add a `%s` placeholder to the string resource.

Reply:
Done with commit 7ef3740